andela/activo-api
# #157767356 Setup project validation framework

#### What does this PR do?
Adds a validation framework to the project using marshmallow.
Marshmallow is a python library that serialises and deserialises data coming into and out of the application while applying validations to them.

#### Description of Task to be completed?
We need a way to easily add validation logic outside the view/controller.
Create a generic validation hook/method that can be applied to any model that is tested with the existing user model.

#### How should this be manually tested?

Clone the branch
Run `pip install -r requrements.txt`
Run `flask shell`
Create a new instance of an asset category and add assets to it by running:
```
from api.models import Asset, AssetCategory
new_category = AssetCategory(name="new category" )
new_category.save()
new_asset=Asset(asset_category_id=new_category.id, tag="abd", serial="def")
new_asset.save()
```
Attempt to delete the new category by running `new_category.delete()`

You should receive the error below:
```
werkzeug.exceptions.Forbidden: 403 Forbidden: Delete failed. The instance has children that are not deleted.
```
#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#157767356](https://www.pivotaltracker.com/story/show/157767356)

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A
